fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/v1/FabricDimensions.java

```diff
@@ -55,9 +53,6 @@ private FabricDimensions() {
 	 */
 	@Nullable
 	public static <E extends Entity> E teleport(E teleported, ServerWorld destination, TeleportTarget target) {
-		Preconditions.checkNotNull(target, "A target must be provided");
-		Preconditions.checkState(!teleported.world.isClient, "Entities can only be teleported on the server side");
-
-		return FabricDimensionInternals.changeDimension(teleported, destination, target);
+		return org.quiltmc.qsl.worldgen.dimension.impl.QuiltDimensionsImpl.teleport(teleported, destination, target);
```

Comment:
Actually, call QuiltDimension's teleport instead; The removal of the Preconditions checks isn't good and I'd ask to add them back, but the public method has them